src/Coupon/SyncerHooks.php

```diff
@@ -299,11 +299,17 @@ protected function handle_delete_coupon( int $coupon_id ) {
 	protected function maybe_send_delete_notification( int $coupon_id ): void {
 		$coupon = $this->wc->maybe_get_coupon( $coupon_id );
 
-		if ( $coupon instanceof WC_Coupon && $this->coupon_helper->should_trigger_delete_notification( $coupon ) ) {
+		if ( is_null( $coupon ) ) {
```

Comment:
I guess since we are also listening to actions `trashed_post` or `deleted_post` for triggering delete coupons function, this line would cause the `coupon.delete` notification being triggered when a product is trashed.

https://github.com/woocommerce/google-listings-and-ads/blob/61fe02dddb7fd2f941dcada20c096499dd05d1a0/src/Coupon/SyncerHooks.php#L134-L135

Reply:
Thanks @ianlin. That is expected.

When the item is trashed is still available in the DB and it won't be null and we schedule the deletion Notification. In case the item is not anymore in DB we notify the deletion directly.

Reply:
Yeah, I understand in this PR it triggers the deletion notifications straightaway. But my question was a bit different: when I tried to delete a **_product_**, the **_coupon_** deletion notification would be triggered as well. It did not happen prior to this PR.